src/core/net/ip6_mpl.cpp

```diff
@@ -33,89 +33,298 @@
 
 #include <common/code_utils.hpp>
 #include <common/message.hpp>
+#include <common/new.hpp>
 #include <net/ip6.hpp>
 #include <net/ip6_mpl.hpp>
 
 namespace Thread {
 namespace Ip6 {
 
+MplBufferedMessageEntry::MplBufferedMessageEntry(Mpl &aMpl, Ip6 &aIp6, TrickleTimer::Handler aHandler):
+    mMpl(aMpl),
+    mMessage(NULL),
+    mRetransmissionTimer(aIp6.mTimerScheduler, aHandler, NULL, this)
+{
+}
+
 Mpl::Mpl(Ip6 &aIp6):
+    mIp6(aIp6),
     mTimer(aIp6.mTimerScheduler, &Mpl::HandleTimer, this),
+    mTimerExpirations(0),
+    mSequence(0),
+    mSeedId(0),
     mMatchingAddress(NULL)
 {
-    memset(mEntries, 0, sizeof(mEntries));
-    mSequence = 0;
-    mSeed = 0;
+    memset(mSeedSet, 0, sizeof(mSeedSet));
+
+    for (uint32_t i = 0; i < kNumBufferedMessageEntries; i++)
+    {
+        uint8_t *memPtr = reinterpret_cast<uint8_t *>(mRawBuffMem) + i * sizeof(MplBufferedMessageEntry);
+        mBufferedMessageSet[i] = new(memPtr) MplBufferedMessageEntry(*this, aIp6, &Mpl::HandleRetransmissionTimer);
```

Comment:
For the sake of completion, can you add an Mpl destructor to delete these please?